src/main/java/lsd/format/json/JsonPrettyPrinter.kt

```diff
@@ -0,0 +1,41 @@
+package lsd.format.json
+
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.exc.MismatchedInputException
+import lsd.format.config.log
+import org.apache.commons.lang3.StringUtils
+import java.util.*
+
+fun indentJson(document: String?): String? {
+    return if (StringUtils.isBlank(document)) {
+        null
+    } else try {
+        indentJsonMap(document)
+    } catch (e: MismatchedInputException) {
+        if (e.message!!.contains("JsonToken.START_ARRAY")) {
+            return try {
+                indentJsonList(document)
+            } catch (ex: JsonProcessingException) {
+                log().trace(ex.message)
```

Comment:
could be debug?

Reply:
I'm not sure about this one because in this case we are trying to guess what type we can deserialise the json string to. It's one of attempts.